src/plugins/memdump/userhook.cpp

```diff
@@ -348,17 +348,26 @@ static bool make_trap(drakvuf_t drakvuf, drakvuf_trap_info* info, hook_target_en
     trap->name = target->target_name.c_str();
     trap->cb = target->callback;
     trap->data = target;
-    trap->breakpoint.lookup_type = LOOKUP_DTB;
-    trap->breakpoint.dtb = info->regs->cr3;
-    trap->breakpoint.addr_type = ADDR_VA;
-    trap->breakpoint.addr = exec_func;
+
+    page_info pageInfo;
+    int ret = vmi_pagetable_lookup_extended(vmi, info->regs->cr3, exec_func, &pageInfo);
+
+    if (ret != VMI_SUCCESS)
+    {
+        goto fail;
+    }
+
+    trap->breakpoint.lookup_type = LOOKUP_NONE;
+    trap->breakpoint.addr_type = ADDR_PA;
+    trap->breakpoint.addr = pageInfo.paddr;
 
     if (drakvuf_add_trap(drakvuf, trap))
     {
         target->trap = trap;
         return true;
     }
 
+fail:
     PRINT_DEBUG("[MEMDUMP-USER] Failed to add trap :(\n");
     return false;
```

Comment:
This path leads to memory leak of `trap` allocated here:
https://github.com/tklengyel/drakvuf/pull/687/files#diff-da50bc7c3b93e99cfbf61ebae9d8fc87R346